pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -387,6 +393,25 @@ func WriteToFile(config clientcmdapi.Config, filename string) error {
 	return nil
 }
 
+func lockFile(filename string) error {
+	// TODO: find a way to do this with actual file locks. Will
+	// probably need seperate solution for windows and linux.
+	f, err := os.OpenFile(lockName(filename), os.O_CREATE|os.O_EXCL, 0)
```

Comment:
https://github.com/docker/docker/blob/a1e319e847f41e648ebede7f9d79999d75bd14c8/vendor/src/github.com/coreos/etcd/pkg/fileutil/lock_windows.go#L41

Only if we don't care about windows.